e2e_context_test.go

```diff
@@ -393,3 +393,160 @@ func TestContextPerRequestWithTimeoutCancelledByContext(t *testing.T) {
 	// expected error should occur
 	assert.True(t, suppressor.expErrorOccurred)
 }
+
+func TestContextPerRequestRetry(t *testing.T) {
+	var m sync.Mutex
+	TimeOutDuration := 5 * time.Minute
+
+	t.Run("not cancelled", func(t *testing.T) {
+		var callCount int
+		var isCtxCancelled bool
+		ctxCancellationSleepDuration := 1 * time.Minute
+		retryDelayDuration := 1 * time.Second
+
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			m.Lock()
+			defer m.Unlock()
+			callCount++
+
+			if callCount > 1 {
+				w.WriteHeader(http.StatusOK)
+				return
+			}
+
+			w.WriteHeader(http.StatusInternalServerError)
+		}))
+		defer ts.Close()
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		go func() {
+			time.Sleep(ctxCancellationSleepDuration)
+			cancel()
+			m.Lock()
+			defer m.Unlock()
+			isCtxCancelled = true
+		}()
```

Comment:
This sleep is always longer then the whole test, so why do we need it at all? I think we can just remove this goroutine from "not cancelled" test.